atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/CachingTransaction.java

```diff
@@ -42,19 +42,21 @@
 import com.palantir.atlasdb.transaction.api.Transaction;
 
 public class CachingTransaction extends ForwardingTransaction {
-
     private static final Logger log = LoggerFactory.getLogger(CachingTransaction.class);
 
+    private static final int DEFAULT_MAX_CACHE_SIZE = 100_000;
+
     final Transaction delegate;
 
     private final LoadingCache<String, ConcurrentMap<Cell, byte[]>> columnTableCache = CacheBuilder.newBuilder()
             .softValues()
+            .maximumSize(DEFAULT_MAX_CACHE_SIZE)
```

Comment:
It doesn't look like this maxSize is going to give you the protection you desire.  The cache is a bit odd in that it associates Strings (which are really tableRefs) to a ConcurrentMap which contains the actual Cell to byte[] (value) mapping.  These ConcurrentMaps are what grow in size, not the columnTableCache itself.